### PR TITLE
Ability to run multiple shard managers

### PR DESCRIPTION
**Pull request by ickerio**
**Please describe the changes this PR makes and why it should be merged:**
With the current shard manager, you can only choose how many shards you have, and they are assuming you'll only be running it on 1 computer. With the PR change, you can set the startShards and endShards that, the shard manager will be dealing with, then on other servers, there can be different shard managers dealing with another large portion of the shards for the bot. This is used commonly with large bots, so workloads can be split between regions and servers. The PR itself can obviously be expanded on very much, with error checking, but I hope the general idea has come across. The following is an example of how you'd configure it. totalShards cannot be left on auto and must be the total shards that all the shard managers will use, you must also set the startShard and endShard. 

Shard manager 1:  Max Shards: 100, startShard: 0, endShard: 24
Shard manager 2:  Max Shards: 100, startShard: 25, endShard: 49
etc up to 100

**Semantic versioning classification:**  
- [Yes] This PR changes the library's interface (methods or parameters added)
- [No] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [No] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by ickerio** [comment]
Just a thought, might be nicer to just pass an array of the shards you want `[0, 1, 2]` instead of an end and start shard

**Comment by iCrawl** [comment]
I would rather not pass 50 shards into that array 😆

**Comment by ickerio** [comment]
Well you could quite easily generate the array, but it was just a thought.

**Comment by iCrawl** [comment]
I was kidding, haha. Yea this would seem a bit more rational, but it takes out the verbosity a bit.

I like the general idea though 👍

File: src/sharding/ShardingManager.js [hunk 1]
```diff
@@ -139,6 +157,15 @@ class ShardingManager extends EventEmitter {
         return;
       }
 
+      if (this.endShard !== 'auto') {
+        Array(this.endShard + 1 - this.startShard).fill().map((x, i) => i + this.startShard)
+          .forEach(i => {
```
Comment (Crowes) [inline src/sharding/ShardingManager.js]: Why not a one-liner?

``.forEach(i => this.createShart(i));``

Comment (devsnek) [inline src/sharding/ShardingManager.js]: you need to change this to a for..of, we don't use forEach in this library by convention.

Comment (1Computer1) [reply src/sharding/ShardingManager.js]: Or just a normal for loop, starting at `startShard` ending at `endShard`.

**Comment by Lewdcario** [comment]
Closing because seemingly dead.

**Pull request closed without merging**